jscomp/reason_outcome_printer/tweaked_reason_oprint.ml

```diff
@@ -251,6 +251,71 @@ and print_simple_out_type ppf =
     Otyp_class (ng, id, tyl) ->
       fprintf ppf "@[%s#%a%a@]" (if ng then "_" else "")
         print_ident id print_typargs tyl
+#if undefined BS_NO_COMPILER_PATCH then         
+  | Otyp_constr ( (Oide_dot ((Oide_dot (Oide_ident "Js", "Internal")),
+                             ("fn" | "meth" as name )) as id) ,
+                 ([Otyp_variant(_,Ovar_fields [ variant, _, tys], _,_); result] as tyl))
+    ->
+      (* Otyp_arrow*)
+      let make tys result =
+        if tys = [] then
+          Otyp_arrow ("", Otyp_constr (Oide_ident "unit", []),result)
+        else
+            match tys with
+          | [ Otyp_tuple tys as single] -> 
+              if variant = "Arity_1" then
+                Otyp_arrow ("", single, result)
+              else 
+                List.fold_right (fun x acc  -> Otyp_arrow("",x,acc) ) tys result
+          | [single] ->
+              Otyp_arrow ("", single, result)
+          | _ -> 
+              raise_notrace Not_found
+      in
+      begin match (make tys result) with
+      | exception _ ->
+          begin 
+            pp_open_box ppf 0;
+            print_typargs ppf tyl;
+            print_ident ppf id;
+            pp_close_box ppf ()
+          end
+      | res ->
+          begin match name  with
+          | "fn" -> 
+              fprintf ppf "@[<0>(%a@ [@bs])@]" print_out_type_1 res
+          | "meth" ->
+              fprintf ppf "@[<0>(%a@ [@bs.meth])@]" print_out_type_1 res
+          | _ -> assert false 
+          end
+      end
+  | Otyp_constr ((Oide_dot (Oide_dot (Oide_ident "Js", "Internal"), "meth_callback" ) as id) ,
+                 ([Otyp_variant(_,Ovar_fields [ variant, _, tys], _,_); result] as tyl))
+    ->
+      let make tys result =
+          match tys with
+          | [ Otyp_tuple tys as single ] -> 
+              if variant = "Arity_1" then Otyp_arrow ("", single, result)
+              else 
+                List.fold_right (fun x acc  -> Otyp_arrow("",x,acc) ) tys result
+          | [single] ->
+              Otyp_arrow ("", single, result)
+          | _ -> 
+              raise_notrace Not_found
+      in
+      begin match (make tys result) with
+      | exception _ ->
+          begin 
+            pp_open_box ppf 0;
```

Comment:
This block of code (open box, print type args, print ident, close box) also appears above, does it make sense to extract it as a function?